test/e2e/pivoting.go

```diff
@@ -296,24 +305,27 @@ func installIronicBMO(inputGetter func() installIronicBMOInput) {
 	cmd.Dir = path
 	cmd.Env = append(env, os.Environ()...)
 
-	outputPipe, er := cmd.StdoutPipe()
-	Expect(er).To(BeNil(), "Cannot get the stdout from the command")
-	errorPipe, er := cmd.StderrPipe()
-	Expect(er).To(BeNil(), "Cannot get the stderr from the command")
-	err := cmd.Start()
-	Expect(err).To(BeNil(), "Failed to deploy Ironic")
-	data, er := io.ReadAll(outputPipe)
-	Expect(er).To(BeNil(), "Cannot get the stdout from the command")
-	if len(data) > 0 {
-		Logf("Output of the shell: %s\n", string(data))
+	stdoutStderr, er := cmd.CombinedOutput()
+	Logf("%s\n", stdoutStderr)
+	Expect(er).To(BeNil(), "Failed to deploy Ironic")
+
+	deploymentNameList := []string{}
+	if input.deployIronic {
+		deploymentNameList = append(deploymentNameList, "-ironic")
 	}
-	errorData, er := io.ReadAll(errorPipe)
-	Expect(er).To(BeNil(), "Cannot get the stderr from the command")
-	err = cmd.Wait()
-	if len(errorData) > 0 {
-		Logf("Error of the shell: %v\n", string(errorData))
+	if input.deployBMO {
+		deploymentNameList = append(deploymentNameList, "-controller-manager")
+	}
+	// Wait for the deployments to become available
+	clientSet := input.ManagementCluster.GetClientSet()
+	for _, name := range deploymentNameList {
+		deployment, err := clientSet.AppsV1().Deployments(input.Namespace).Get(ctx, input.NamePrefix+name, metav1.GetOptions{})
+		Expect(err).To(BeNil(), "Unable to get the deployment %s in namespace %s \n error message: %s", input.NamePrefix+name, input.Namespace, err)
+		framework.WaitForDeploymentsAvailable(ctx, framework.WaitForDeploymentsAvailableInput{
+			Getter:     input.ManagementCluster.GetClient(),
+			Deployment: deployment,
+		}, input.E2EConfig.GetIntervals(input.SpecName, "wait-deployment")...)
```

Comment:
I am thinking if this waiting part should be added in a function?  What if when we scale down and up ironic/bmo deployment and we want to check /wait until deployment is available or not ? That workflow wont necessarily install ironic and bmo.

Reply:
And this could be then generic function for any deployment/deployments you would want to wait to be available .

Reply:
We are already using `framework.WaitForDeploymentsAvailable` which is defined in the framework I do think we have to wrap this function call in another function

Reply:
ow, then why are we putting deployment names in a list, then running a loop to get an individual deployment name again from that list and then calling this function. Why dont we just call the function with the deployment name directly.

Reply:
I do not know why CAPI call this function `WaitForDeploymentsAvailable` although it accept only one deployment to wait for! since I am not sure if the install function is deployment only ironic or bmo or both at the same call I am doing this check and looping over one improvement I can do is to get the list of all deployment in the namespace and loop over it but I cannot be should that this name space only contains bmo and ironic so it is safer to check with the name

Reply:
Ok I get the issue is the uncertainity if the install function is called for installing bmo, ironic or both. In that case , instead of putting the name of the deployment in the list in the 
`if input.deployBMO {`  we might actually call the `framework.WaitForDeploymentsAvailable` function already for the deployment. This would be a good improvement for the future. For now I am happy to take this in so that we have proper workflow atleast.
/approve